### ISSUE TITLE
Pls detail arduino 2.1.0 Tools config settings to build camera.ino

### ISSUE THREAD

**Issue by achalmers** [m5stack/M5CoreS3#30]
Have been ableto build all examples except Camera (required updating arduino IDE and many dependent libraries). Spent several days on the Camera.ino playing with Tools options in Arduino2.1.0 but still getting weird serial port failure errors reported at 75% programmed .
I noticed the partitions.csv is included in the Camera directory. Im guessing it gets used automatically,  im wondering if it is correct also ? (Ive seen errors "No core dump partition found!")  Heres the original csv.
[partitions.csv](https://github.com/m5stack/M5CoreS3/files/11667543/partitions.csv)

Also noticed no sample code for touchscreen, or the very nice factory application.

**Comment by achalmers** [comment]
Applying the 2 new firmware updates fixed the serial port error on download (I also slowed down burningto115200). I suspect the csv needs a line for a coredump partition to be added in order to debug the (last?) runtime issue. I also see the FactoryApp code was added to the repo for PlatformIO.

**Comment by IcingTomato** [comment]
Would you like to provide more information about camera demo serial output message on Arduino 2.1.0?

**Comment by achalmers** [comment]
Before applying the 2 firmware updates (see https://docs.m5stack.com/en/core/CoreS3 ) the Camera.ino would compile but during the program stage it would stop at 75% complete and generate a 'serial port error'.


After applyingthose hotfixes/firmare updates I now am able to build the Camera.ino and program the device, but when it starts execution the serial output shows :   
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x15 (USB_UART_CHIP_RESET),boot:0x28 (SPI_FAST_FLASH_BOOT)
Saved PC:0x42092d76
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x3ac
load:0x403c9700,len:0x9bc
load:0x403cc700,len:0x27c0
entry 0x403c98c0
E (206) esp_core_dump_flash: No core dump partition found!
E (206) esp_core_dump_flash: No core dump partition found!

Im guessing I havea Tool/Setting incorrect as well as no dump partition in the partitions_ffat.csv   .

**Comment by IcingTomato** [comment]
![242071104-ff9b8a9b-e548-4a4a-9ec8-b8cc64563b35](https://github.com/m5stack/M5CoreS3/assets/24474531/28c72e1d-fe55-4de0-a42e-00318e67f7cc)
Would you like to open PSRAM and try one more time?

**Comment by IcingTomato** [comment]
I uploaded a PlatformIO demo just now. 
https://github.com/m5stack/M5CoreS3/tree/6e0c70bd913a4a8bc98fcc861652cb3c5aba882b
You can have a try.

**Comment by achalmers** [comment]
Yes PSRAM was set see Settings below.   Serial Output also pasted
![image](https://github.com/m5stack/M5CoreS3/assets/7535369/85c61637-57ed-4b88-b259-4742e640b04f)

![image](https://github.com/m5stack/M5CoreS3/assets/7535369/be4dee6d-f7fd-4adc-b248-f1bb938fee0b)

**Comment by IcingTomato** [comment]
Strangely enough, could you try the PlatformIO camera example?
https://github.com/m5stack/M5CoreS3/tree/main/examples/PlatformIO/Camera
I cannot reproduce your mistakes. Would you like to erase all the flash and re-upload the codes?
![image](https://github.com/m5stack/M5CoreS3/assets/24474531/6dcc76d0-af52-4dc8-879e-17d3c8116fcb)

**Comment by achalmers** [comment]
Yes I did erase flash.

Thx for adding the Platformio project. 
OK, Ive been holding off updating to Platformio but I guess its time.
I downloaded the latest Platformio and loaded camera project and required libraries, the camera.ini build seems to halt when freeRTOS (called by esp_camera ?)  needs a setting, see below for error. Not sure what to set or how to define (eg add --device xxxx  to its line in the library section ?). Probably due tome not using Platformio before. 

![image](https://github.com/m5stack/M5CoreS3/assets/7535369/8a75d8f4-571c-4da8-9425-76fafa9ed67c)

**Comment by IcingTomato** [comment]
Oh, I see.
You should pick the demo document out of the library.
![image](https://github.com/m5stack/M5CoreS3/assets/24474531/5923f18d-5de0-4784-a8ac-267d35a87af5)
